code/game/gamemodes/cult/runes.dm

```diff
@@ -359,8 +229,20 @@ var/list/teleport_runes = list()
 	var/list/myriad_targets = list()
 	var/turf/T = get_turf(src)
 	for(var/mob/living/M in T)
-		if(!iscultist(M))
+		if(!iscultist(M) || (M.mind && is_sacrifice_target(M.mind)))
 			myriad_targets |= M
+
+	for(var/obj/item/organ/external/head/head in T)
+		var/obj/item/organ/internal/brain/brain = locate(/obj/item/organ/internal/brain) in head
+		if(brain)
+			var/mob/living/carbon/brain/B = locate(/mob/living/carbon/brain) in brain
+			if(B.mind && is_sacrifice_target(B.mind))
+				myriad_targets |= B
```

Comment:
Same here. Dupes should not be allowed due to your for loops. Same for the one below